### PR TITLE
feat: adds transition shorthand

### LINKED ISSUE #177: shorthands: create a function for transition property
transition: ".3s cubic-bezier(.33,0,0,1) all" can't be used with griffel atm

### PR DESCRIPTION
**Pull request by bsunderhus**
Fixes #177

### CODE REVIEW

File: change/@griffel-core-cca9a4d2-a460-4d77-a750-b6dd381c37f5.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "feat: adds transition shorthand",
```
Comment (layershifter) [inline change/@griffel-core-cca9a4d2-a460-4d77-a750-b6dd381c37f5.json:3]:
```suggestion
  "comment": "feat: add transition shorthand",
```

nit

File: packages/core/src/shorthands/types.ts [hunk 2]
```diff
@@ -24,3 +24,9 @@ export type PaddingInput = ValueOrArray<CSS.Property.Padding<GriffelStylesCSSVal
 export type InsetInput = ValueOrArray<CSS.Property.Inset<GriffelStylesCSSValue>>;
 
 export type FlexInput = [CSS.Property.Flex, CSS.Property.Flex?, CSS.Property.Flex?];
+
+export type TransitionPropertyInput = CSS.Property.TransitionProperty;
+export type TransitionDurationInput = CSS.Property.TransitionDuration;
+export type TransitionDelayInput = CSS.Property.TransitionDelay;
+export type TransitionTimingFunctionInput = CSS.Property.TransitionTimingFunction;
+export type TransitionGlobalnInput = CSS.Globals;
```
Comment (layershifter) [inline packages/core/src/shorthands/types.ts:32]:
```suggestion
export type TransitionGlobalInput = CSS.Globals;
```

Oops, a typo

File: packages/core/src/shorthands/transition.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    {},
+  );
+}
+
+type TransitionInputs = [
+  TransitionPropertyInput,
+  TransitionDurationInput?,
+  TransitionDelayInput?,
+  TransitionTimingFunctionInput?,
+];
```
Comment (layershifter) [inline packages/core/src/shorthands/transition.ts:81]: nit: let's move `TransitionInputs` before `TransitionStyle`

**Review by layershifter — APPROVED** [review]
Small nits, other than that LGMT 👍

**Review by ling1726 — CHANGES_REQUESTED** [review]
Please add some docs for this

**Comment by ling1726** [comment]
@bsunderhus  can you add documentation to the docsite ? https://github.com/microsoft/griffel/blob/main/apps/website/docs/react/api/shorthands.md

**Comment by bsunderhus** [comment]
> @bsunderhus can you add documentation to the docsite ? https://github.com/microsoft/griffel/blob/main/apps/website/docs/react/api/shorthands.md

So many documentation files 🙈, apparently, I've missed that one! thx @ling1726

**Review by ling1726 — APPROVED** [review]

**Pull request merged**